functions/src/contactCreated/index.js

```diff
@@ -0,0 +1,96 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import { to } from 'utils/async';
+
+const eventName = 'contactCreated';
+
+/**
+ * Emails user's who's UIDs are in the system_settings/notifications
+ * @param {admin.firestore.DataSnapshot} snap - Data snapshot of the event
+ * @param {Function} snap.data - Value of document
+ * @param {functions.EventContext} context - Function event context
+ * @param {object} context.auth - Authentication information for the user that triggered the function
+ * @returns {Promise} Resolves with null after handling contact creation
+ */
+async function contactCreatedEvent(snap, context) {
+  const { params, auth, timestamp } = context;
+  console.log('contactCreated onCreate event:', snap.data());
+
+  // Load settings doc
+  const settingsRef = admin.firestore().doc('system_settings/notifications');
+  const [settingsDocErr, settingsDocSnap] = await to(settingsRef.get());
+
+  // Handle errors loading settings docs
+  if (settingsDocErr) {
+    console.error(
+      `Error loading settings doc: ${settingsDocErr.message || ''}`,
+      settingsDocErr,
+    );
+  }
+
+  // Notify all users in newRequests parameter of system_settings/notification document
+  const { newContacts: userUids } = settingsDocSnap.data() || {};
+
+  // Exit if there are no uids
+  if (!userUids) {
+    console.log('No user uids found for new contacts, exiting...');
+    return null;
+  }
+
+  const [getUserEmailErrors, userEmails] = await to(
```

Comment:
Not a big deal for now, but, FYI, [the extension allows](https://github.com/firebase/extensions/tree/master/firestore-send-email) setting toUids to accomplish this.

```
Users collection: A collection of documents keyed by user UID. If the toUids, ccUids, and/or bccUids recipient options are used in the added email document, this extension delivers email to the email field based on lookups in this collection.
```

Reply:
Thanks for pointing out, TIL! Assuming that this uses the email on the auth user instead of from the profile, so I'm going to make sure to add logic to the account page that also updates the auth email if the user is changing